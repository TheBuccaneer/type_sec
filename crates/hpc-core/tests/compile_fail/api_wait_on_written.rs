use hpc_core::*;

fn main() {
    let ctx = Context::create_context().unwrap();
    let queue = ctx.create_queue().unwrap();

    // Buffer → Ready
    let buf = ctx
        .create_empty_buffer::<u8>(16).unwrap()
        .write_block(&queue, &[0u8; 16]).unwrap();

    // Kernel starten → InFlight + Event
    let (inflight, _evt) = buf.enqueue_kernel(&queue, &unimplemented!(), 16).unwrap();

    let mut out = [0u8; 16];

    // Verboten: Read auf InFlight
    inflight.write_blocking(&queue, &mut out).unwrap();
}
