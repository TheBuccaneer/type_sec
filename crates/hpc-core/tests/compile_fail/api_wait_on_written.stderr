error[E0599]: no method named `write_blocking` found for struct `DeviceBuffer<'_, u8, InFlight>` in the current scope
  --> tests/compile_fail/api_wait_on_written.rs:19:14
   |
19 |     inflight.write_blocking(&queue, &mut out).unwrap();
   |              ^^^^^^^^^^^^^^ method not found in `DeviceBuffer<'_, u8, InFlight>`
   |
   = note: the method was found for
           - `DeviceBuffer<'brand, T, Mapped>`
           - `DeviceBuffer<'brand, T, Written>`
